ocrd_eval.md

```diff
@@ -0,0 +1,372 @@
+
+Example:
+![A Fraktur sample reading "Sonnenfinſterniſſe:"](https://pad.gwdg.de/uploads/d7fa6f23-7c79-4fb2-ad94-7e98084c69d6.jpg)
+
+This reads `Sonnenfinſterniſſe:`. The output contains `Sonnenfinſterniſſe`, deleting `:`.
+
+* **substitution**: a character is replaced by another character in the output.
+
+Example:
+
+![A Fraktur sample reading "Die Finſterniſſe des 1801ſten Jahrs"](https://pad.gwdg.de/uploads/b894049b-8d98-4fe7-ac31-71b2c9393a6c.jpg)
+
+This heading reads `Die Finſterniſſe des 1801ſten Jahrs`. The output contains `180iſten`, replacing `1` with `i`.
+
+* **insertion**: a new character is introduced in the output.
+
+Example:
+
+![A Fraktur sample reading "diese Strahlen, und"](https://pad.gwdg.de/uploads/e6b6432e-d79c-4568-9aef-15a026c05b39.jpg)
+
+This reads `diese Strahlen, und`. The output contains `Strahlen ,`, inserting a white space before the comma.
+
+CER can be calculated in several ways, depending on whether a normalized CER is used or not.
+
+Given $i$ as the number of insertions, $d$ the number of deletions, $s$ the number of substitutions and $n$ the total number of characters in a text, the CER can be obtained by
+
+$CER = \frac{i + s+ d}{n}$
+
+If the CER value is calculated this way, it represents the percentage of characters incorrectly recognized by the OCR engine. Also, we can easily reach error rates beyond 100% when the output contains a lot of insertions.
+
+The *normalized* CER tries to mitigate this effect by considering the number of correct characters, $c$:
+
+$CER_n = \frac{i + s+ d}{i + s + d + c}$
+
+In OCR-D's benchmarking we calculate the *non-normalized* CER where values over 1 should be read as 100%.
+
+###### CER Granularity
+
+In OCR-D we distinguish between the CER per **page** and the **overall** CER of a text. The reasoning behind this is that the material OCR-D mainly aims at (historical prints) is very heterogeneous: Some pages might have an almost simplistic layout while others can be highly complex and difficult to process. Providing only an overall CER would cloud these differences between pages.
+
+At this point we only provide a CER per page; an overall CER might be calculated as a weighted aggregate at a later stage.
+
+##### Word Error Rate (WER)
+
+The word error rate (WER) is closely connected to the CER. While the CER focusses on differences between characters, the WER represents the percentage of words incorrectly recognized in a text.
+
+CER and WER share categories of errors, and the WER is similarly calculated:
+
+$WER = \frac{i_w + s_w + d_w}{n_w}$
+
+where $i_w$ is the number of inserted, $s_w$ the number of substituted, $d_w$ the number of deleted and $n_w$ the total number of words.
+
+More specific cases of WER consider only the "significant" words, omitting e.g. stopwords from the calculation.
+
+###### WER Granularity
+
+In OCR-D we distinguish between the WER per **page** and the **overall** WER of a text. The reasoning here follows the one of CER granularity.
```

Comment:
I might have a terminology problem here: What is the difference between a "document" and the "entire corpus of a text"? With document I always had something like "the complete monograph / digitization of Goethe's Faust in edition ABC published in XYZ by ASDF" in mind.

But it my terminology might be confusing because the term "document" might also refer to a single digitized page (which would, in turn, raise the same question of the relation between "document" and "page".)

Reply:
Yes, document should be a work (even if partial), but in our evaluation we usually cover a collection of documents, hence the term corpus.

Reply:
Thanks for the explanation. I'm a bit hesitant about introducing the corpus level here because this (could) imply that we offer a WER for a complete corpus, too. Up to now we haven't discussed this.

Ping @cneud @kba

Reply:
@cneud, @kba and I agreed that while this is a valid point the corpus level is currently out of scope for what we're planning in the current phase.

(We kind of have it, indirectly, since the sample "works" are actually merges of pages taken from several different works.)

Reply:
`Resolved` then?

Reply:
Yes!